chess/gaviota.py

```diff
@@ -179,26 +166,15 @@ def init_ppp48_idx() -> Tuple[List[List[List[int]]], List[int], List[int], List[
     for x in range(48):
         for y in range(x + 1, 48):
             for z in range(y + 1, 48):
-                a = ITOSQ[x]
-                b = ITOSQ[y]
-                c = ITOSQ[z]
-                if not in_queenside(b) or not in_queenside(c):
+                if not (in_queenside(ITOSQ[y]) and in_queenside(ITOSQ[z])):
                     continue
 
-                i = a - 8
-                j = b - 8
-                k = c - 8
+                i, j, k = ITOSQ[x] - 8, ITOSQ[y] - 8, ITOSQ[z] - 8
 
                 if idx_is_empty(ppp48_idx[i][j][k]):
-                    ppp48_idx[i][j][k] = idx
-                    ppp48_idx[i][k][j] = idx
-                    ppp48_idx[j][i][k] = idx
-                    ppp48_idx[j][k][i] = idx
-                    ppp48_idx[k][i][j] = idx
-                    ppp48_idx[k][j][i] = idx
-                    ppp48_sq_x[idx] = i
-                    ppp48_sq_y[idx] = j
-                    ppp48_sq_z[idx] = k
+                    for p in [(i, j, k), (i, k, j), (j, i, k), (j, k, i), (k, i, j), (k, j, i)]:
+                        ppp48_idx[p[0]][p[1]][p[2]] = idx
```

Comment:
You can use [`itertools.permutations()`](https://docs.python.org/3/library/itertools.html#itertools.permutations) to simplify this loop.
```python
for ii, jj, kk in itertools.permutations((i, j, k)):
    ppp48_idx[ii][jj][kk] = idx
```

Reply:
That's correct, thanks for the idea.